rfcs/0004-bp-go-build-ldflags.md

```diff
@@ -0,0 +1,26 @@
+# BP_GO_BUILD_LDFLAGS
+
+## Proposal
+
+Add an environment variable `BP_GO_BUILD_LDFLAGS` which would allow users to
+set the value of `-ldflags` for their `go build` command.
+
+## Motivation
+
+The `-ldflags` flag can have complicated values with multiple sets of internal
+quotation marks, an example of this complication can be seen in [this
+issue](https://github.com/paketo-buildpacks/go-build/issues/129). The goal of
+this environment variable is to remove one quotation level hopefully making it
+easier to express the `-ldflags` value with little to no escaping.
```

Comment:
It would be useful to see the diff in what the user experience would be like given the same example from my github issue, i.e. showing how hard it is to just use the `BP_GO_BUILD_...` env var and then being able to use this flag instead.


```diff
- BP_GO_BUILD_FLAGS='-ldflags="-extldflags -f no-PIC -static" -tags=osusergo,netgo,embedfs'
+ BP_GO_BUILD_LDFLAGS="-extldflags -f no-PIC -static" -tags=osusergo,netgo,embedfs"
```

Reply:
I added a small example that hopefully gets the change across. I would love any feedback you have on that example though!